daffodil-sapi/src/main/scala/org/apache/daffodil/sapi/packageprivate/Utils.scala

```diff
@@ -71,6 +71,8 @@ private[sapi] object ValidationConversions {
       case ValidationMode.Off => SValidationMode.Off
       case ValidationMode.Limited => SValidationMode.Limited
       case ValidationMode.Full => SValidationMode.Full
+      // todo;; hacking this out for now
+      case _ => SValidationMode.Off
```

Comment:
Not sure how to approach this

Reply:
Grrr. I am not sure why SAPI can't just directly use the api.ValidationMode class, and has to impose its own Enumeration. 

So you can do a parallel change to what has to be done for JAPI here, i.e., add extra args to sapi.setValidationMode and withValidationMode, and use them to construct a api.ValidationMode.Custom object once you get that point.

Reply:
I've alway sort of viewed ``.api`` as a sort of internal API and ``.sapi`` or ``.japi`` as external publicly usble APIs. For example, right now we only generate java/scala docs for for the daffodil-japi/sapi projects, so anything not in those projects will not be documented. So our sapi/japi sort of wrap things in api.

Maybe this modeToScala thing was a way to get what the internal API though the validation mode was and return it as something an sapi user would understand. But it sounds like we don't actually have getters for the current validation mode, so this is never used?